app/questionnaire/navigation.py

```diff
@@ -40,14 +40,25 @@ def build_navigation(self, current_group_id, current_group_instance):
             logger.debug("Building frontend navigation for group %s", group['id'])
 
             repeating_rule = SchemaHelper.get_repeat_rule(group)
+            skip_group = self._should_skip_group(current_group_instance, group)
+
+            if not skip_group:
```

Comment:
Unsure whether the alternative reads better, looks odd the skip_group appearing twice
```
if repeating_rule and not skip_group:
                    navigation.append(self._build_repeating_navigation(repeating_rule, group, current_group_id, current_group_instance))
                elif not skip_group:
                    navigation.append(self._build_single_navigation(group, current_group_id, first_location))```